app/services/invoices/payments/adyen_service.rb

```diff
@@ -0,0 +1,178 @@
+        )
+      end
+
+      def adyen_payment_provider
+        @adyen_payment_provider ||= organization.adyen_payment_provider
+      end
+
+      def update_payment_method_id
+        result = client.checkout.payments_api.payment_methods(payment_method_params).response
+
+        if (payment_method_id = result['storedPaymentMethods']&.first&.dig('id'))
+          customer.adyen_customer.update! payment_method_id: payment_method_id
+        end
+      rescue Adyen::AdyenError => e
+        deliver_error_webhook(e)
+        raise
+      end
+
+      def create_adyen_payment
+        update_payment_method_id
+
+        client.checkout.payments_api.payments(payment_params).response
+      rescue Adyen::AdyenError => e
+        deliver_error_webhook(e)
+        update_invoice_payment_status(payment_status: :failed, deliver_webhook: false)
+
+        raise
+      end
+
+      def payment_method_params
+        {
+          merchantAccount: adyen_payment_provider.merchant_account,
+          shopperReference: customer.external_id,
+        }
+      end
+
+      def payment_params
+        prms = {
+          amount: {
+            currency: invoice.currency.upcase,
+            value: invoice.total_amount_cents
+          },
+          reference: invoice.number,
+          paymentMethod: {
+            type: "scheme",
+            storedPaymentMethodId: customer.adyen_customer.payment_method_id
+          },
+          shopperReference: customer.external_id,
+          merchantAccount: adyen_payment_provider.merchant_account,
+          shopperInteraction: "ContAuth",
+          recurringProcessingModel: "UnscheduledCardOnFile"
+        }
+        prms[:shopperEmail] = customer.email if customer.email
+        prms
+      end
+
+      def invoice_payment_status(payment_status)
+        return :pending if PENDING_STATUSES.include?(payment_status)
+        return :succeeded if SUCCESS_STATUSES.include?(payment_status)
+        return :failed if FAILED_STATUSES.include?(payment_status)
+
+        payment_status
+      end
+
+      def update_invoice_payment_status(payment_status:, deliver_webhook: true)
+        result = Invoices::UpdateService.call(
+          invoice:,
+          params: {
+            payment_status:,
+            ready_for_payment_processing: payment_status.to_sym != :succeeded,
+          },
+          webhook_notification: deliver_webhook,
+        )
+        result.raise_if_error!
+      end
+
+      def increment_payment_attempts
+        invoice.update!(payment_attempts: invoice.payment_attempts + 1)
+      end
+
+      def deliver_error_webhook(adyen_error)
+        return unless invoice.organization.webhook_url?
+
+        SendWebhookJob.perform_later(
+          'invoice.payment_failure',
+          invoice,
+          provider_customer_id: customer.adyen_customer.provider_customer_id,
+          provider_error: {
+            message: adyen_error.msg,
+            error_code: adyen_error.code
+          }
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)